lib/puppet/type/sensu_check.rb

```diff
@@ -21,10 +21,13 @@ def should
   def initialize(*args)
     super *args
 
-    self[:notify] = [
-      "Service[sensu-client]",
-      "Service[sensu-server]",
-    ].select { |ref| catalog.resource(ref) }
+    if c = catalog
+      self[:notify] = [
+        "Service[sensu-client]",
+        "Service[sensu-server]",
+        "Service[sensu-enterprise]",
```

Comment:
@ghoneycutt I modified the provisioning manifest for `sensu-server-enterprise` to manage Service[sensu-enterprise] instead of Service[sensu-server].  Take a look at this change in 1d694e4, I assumed `tests/sensu-server-enterprise.pp` is intended to setup sensu-enterprise.

With `vagrant up sensu-server`, the notification doesn't cause a problem as we expect.  With `vagrant up sensu-server-enterprise`, the notification does properly notify `Service[sensu-enterprise]' when a managed `sensu::check` resource changes state.